### ISSUE TITLE
Throttle with timeout is producing infinite stack traces

### ISSUE THREAD

**Issue by mikila85** [ReactiveX/rxjs#4963]
## Bug Report

**Current Behavior**

Throttle with timeout (becuase i need leading & trailing) is creating a very long stack trace of repeated setTimeouts:

![image](https://user-images.githubusercontent.com/3965957/62863067-6d286100-bd10-11e9-9121-bbf8f31b9bdb.png)

**Reproduction**
no reproduction, need to debug stack trace.

```js
this.updateSubject.pipe(throttle(() => interval(5000), {
            leading: true,
            trailing: true
        })).subscribe(() => this.doAction());
```

**Expected behavior**
should not create larger and larger stack traces

**Environment**
- Runtime: Chrome latest
- RxJS version: 6.3.3

**Comment by cartant** [comment]
You probably want to use `timer` instead of `interval` - the former will emit only once, whereas the latter will emit repeatedly.

**Comment by mikila85** [comment]
but i want it to happen every 5000 miliseconds, repeatedly..

**Comment by cartant** [comment]
From the docs:

> Emits a value from the source Observable, then ignores subsequent source values for a duration determined by **another Observable**, then **repeats this process**.

Where '**another Observable**' is the the observable you return from the function passed to `throttle`. Note that the function is called **multiple times**; it's called **each time** a value is emitted from the throttled observable - that's what's implied by '**repeats this process**'.

**Comment by mikila85** [comment]
After much fiddling i found the source of the problem,
it doesn't matter if you use timer or interval, the problem is with setting` {trailing: true}`
here is a [demo (keep F12 open to see the stack in the debugger)](https://stackblitz.com/edit/jxwswz?file=index.ts)

if you remove the trailing: true, the issue doesn't reproduce.

**Comment by mikila85** [comment]
You are right, timer is the one I should have used, thank you for pointing it out to me! 
Still when combined with leading and trailing I see stack trace getting longer and longer :(

**Comment by cartant** [comment]
Yeah, I suspect this is actually a bug. It should not matter whether or not the notifier emits multiple times - the `throttle` implementation should unsubscribe upon the first notification - but, unfortunately, that does not appear to be happening. I will have a look at this sometime - so please leave this open - but, in general I'd suggest that it's best to ensure any notifier emits only once.
